hcloud/error.go

```diff
@@ -11,20 +11,36 @@ const (
 	ErrorCodeLimitReached           = "limit_reached" // Rate limit reached
 	ErrorCodeUnknownError           = "unknown_error" // Unknown error
 	ErrorCodeNotFound               = "not_found"     // Resource not found
+	ErrorCodeInvalidInput           = "invalid_input" // Validation error
 )
 
 // Error is an error returned from the API.
 type Error struct {
 	Code    ErrorCode
 	Message string
+	Details interface{}
 }
 
 func (e Error) Error() string {
 	return fmt.Sprintf("%s (%s)", e.Message, e.Code)
 }
 
+// ErrorInvalidInputDetails contains the details of an 'invalid_input' error.
+type ErrorInvalidInputDetails struct {
+	Fields []*ErrorInvalidInputDetailsField `json:"fields"`
```

Comment:
Structs in root `hcloud` package should not have `json` tags.

Reply:
I would name it `ErrorDetailsInvalidInput` so all details structs will have the same prefix `ErrorDetails`.

Reply:
Also, any particular reason why it’s a pointer and not simply a value? As a rule of thumb, I usually use values for things that don’t have an identity.